### ISSUE TITLE
Improve the report location of `arrow-paren`

### ISSUE THREAD

**Issue by golopot** [eslint/eslint#11773]
**What rule do you want to change?**
`arrow-paren`

Currently the reported node is the whole function. It makes sense to change the reported node to the parameter node.

```js
/* eslint arrow-parens: [2] */

// current report location
const f = a => a + a + a;
//        ~~~~~~~~~~~~~~ Error: needs parens.

// proposed report location
const f = a => a + a + a;
//        ~              Error: needs parens.

```
[demo link](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IGFycm93LXBhcmVuczogWzJdICovXG5cbmNvbnN0IGYgPSBhID0+IGEgKyBhICsgYTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7Im5vLXVuZGVmIjoyLCJuby10aHJvdy1saXRlcmFsIjoyfSwiZW52Ijp7fX19)

**Are you willing to submit a pull request to implement this change?**
yes

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. This seems like a worthwhile change to me.

**Comment by platinumazure** [comment]
I agree this should be done. I'll champion this change.

@eslint/eslint-team This needs one more :+1: to be accepted.

**Comment by not-an-aardvark** [comment]
This is now accepted.
